gemrb/core/DisplayMessage.h

```diff
@@ -63,46 +57,49 @@ class GEM_EXPORT DisplayMessage
 	static bool EnableRollFeedback();
 	static String ResolveStringRef(ieStrRef);
 	
-	template<typename ...ARGS>
-	void DisplayRollStringName(const String& fmt, const Color &color, const Scriptable *speaker, ARGS&& ...args) const {
-		String formatted = fmt::format(fmt, std::forward<ARGS>(args)...);
-		DisplayStringName(std::move(formatted), color, speaker);
-	}
+	/** returns the speaker's color and name */
+	Color GetSpeakerColor(String& name, const Scriptable *&speaker) const;
+	
 public:
 	static ieStrRef GetStringReference(size_t);
 	static bool HasStringReference(size_t);
 
 public:
 	DisplayMessage();
-	/** returns the speaker's color and name */
-	Color GetSpeakerColor(String& name, const Scriptable *&speaker) const;
 	/** displays any string in the textarea */
-	void DisplayMarkupString(String txt) const;
+	void DisplayMarkupString(String txt) const;	
 	/** displays a string constant in the textarea */
-	void DisplayConstantString(size_t stridx, const Color &color, Scriptable *speaker = nullptr) const;
-	/** displays actor name - action : parameter */
-	void DisplayConstantStringNameString(size_t stridx, const Color &color, size_t stridx2, const Scriptable *actor) const;
+	void DisplayConstantString(size_t stridx, unsigned char color, Scriptable *speaker = nullptr) const;
 	/** displays a string constant followed by a number in the textarea */
-	void DisplayConstantStringValue(size_t stridx, const Color &color, ieDword value) const;
-	/** displays a string constant in the textarea, starting with speaker's name */
+	void DisplayConstantStringValue(size_t stridx, unsigned char color, ieDword value) const;	
+	/** displays actor name - action : parameter */
+	void DisplayConstantStringName(size_t stridx, unsigned char color, const Scriptable *speaker) const;
 	void DisplayConstantStringName(size_t stridx, const Color &color, const Scriptable *speaker) const;
-	/** displays a string constant in the textarea, starting with speaker's name, also replaces one numeric value (it is a format string) */
-	void DisplayConstantStringNameValue(size_t stridx, const Color &color, const Scriptable *speaker, int value) const;
 	/** displays a string constant in the textarea, starting with actor, and ending with target */
-	void DisplayConstantStringAction(size_t stridx, const Color &color, const Scriptable *actor, const Scriptable *target) const;
+	void DisplayConstantStringAction(size_t stridx, unsigned char color, const Scriptable *actor, const Scriptable *target) const;	
+	/** displays a string in the textarea, starting with speaker's name */
+	void DisplayStringName(ieStrRef stridx, unsigned char color, const Scriptable *speaker, STRING_FLAGS flags) const;
+	void DisplayStringName(String text, unsigned char color, const Scriptable *speaker) const;
+	void DisplayStringName(String text, const Color &color, const Scriptable *speaker) const;
```

Comment:
can this function not be private?

Reply:
The git grep show this:
```
gemrb/core/EffectQueue.cpp:                     displaymsg->DisplayStringName(globals.Opcodes[fx->Opcode].Strref, DMC_WHITE, target, STRING_FLAGS::SOUND);
gemrb/core/GameScript/GSUtils.cpp:                                      displaymsg->DisplayStringName(Strref, DMC_WHITE, Sender, STRING_FLAGS::NONE);
gemrb/core/Map.cpp:     displaymsg->DisplayStringName(trackString, DMC_LIGHTGREY, target, STRING_FLAGS::NONE);
gemrb/core/Projectile.cpp:                      displaymsg->DisplayStringName(StrRef, DMC_LIGHTGREY, act, STRING_FLAGS::NONE);
gemrb/core/Scriptable/Actor.cpp:                        displaymsg->DisplayStringName(std::move(msg), DMC_WHITE, this);
gemrb/core/Scriptable/Actor.cpp:                        displaymsg->DisplayStringName(msg + dmg, DMC_WHITE, this);
gemrb/core/Scriptable/Actor.cpp:                        displaymsg->DisplayStringName(ModalStates[Modal.State].leaving_str, DMC_WHITE, this, STRING_FLAGS::SOUND | STRING_FLAGS::SPEECH);
gemrb/core/Scriptable/Actor.cpp:                displaymsg->DisplayStringName(std::move(rollLog), DMC_WHITE, this);
gemrb/core/Scriptable/Actor.cpp:                                        displaymsg->DisplayStringName(ModalStates[Modal.State].entering_str, DMC_WHITE, this, STRING_FLAGS::SOUND | STRING_FLAGS::SPEECH);
gemrb/core/Scriptable/Actor.cpp:                                        displaymsg->DisplayStringName(ModalStates[Modal.State].failed_str, DMC_WHITE, this, STRING_FLAGS::SOUND | STRING_FLAGS::SPEECH);
gemrb/core/Scriptable/Actor.cpp:                        displaymsg->DisplayStringName(core->GetString(ieStrRef::MD_SUCCESS), DMC_WHITE, this);
gemrb/core/Scriptable/Actor.cpp:                displaymsg->DisplayStringName(core->GetString(ieStrRef::MD_FAIL), DMC_WHITE, this);
gemrb/core/Scriptable/Container.cpp:                    displaymsg->DisplayStringName(OpenFail, DMC_BG2XPGREEN, actor, STRING_FLAGS::SOUND | STRING_FLAGS::SPEECH);
gemrb/core/Scriptable/Door.cpp:                 displaymsg->DisplayStringName(OpenStrRef, DMC_BG2XPGREEN, actor, STRING_FLAGS::SOUND | STRING_FLAGS::SPEECH);
gemrb/core/Scriptable/Scriptable.cpp:                   displaymsg->DisplayStringName(std::move(str), DMC_WHITE, this);
gemrb/core/Scriptable/Scriptable.cpp:                   displaymsg->DisplayStringName(std::move(str), DMC_WHITE, this);
gemrb/core/Scriptable/Scriptable.cpp:                   displaymsg->DisplayStringName(s1 + L" " + s2, DMC_WHITE, this);
gemrb/plugins/FXOpcodes/FXOpcodes.cpp:          displaymsg->DisplayStringName(ieStrRef(fx->Parameter1), DMC_WHITE, target, STRING_FLAGS::SOUND | STRING_FLAGS::SPEECH);
gemrb/plugins/GUIScript/GUIScript.cpp:          displaymsg->DisplayStringName(strref, ColorFromPy(pycol), actor, STRING_FLAGS::SOUND);
```
Is used in a log of places, the worst is **ColorFromPy(pycol)**

Reply:
I was able to put this one private:
```
void DisplayStringName(String text, const Color &color, const Scriptable *speaker) const;
```
The rest where needed
```
[build]~/gemrb-myfork-git/gemrb/core/DialogHandler.cpp: In member function ‘bool GemRB::DialogHandler::DialogChoose(unsigned int)’:
[build]~/gemrb-myfork-git/gemrb/core/DialogHandler.cpp:345:146: error: ‘void GemRB::DisplayMessage::DisplayStringName(GemRB::ieStrRef, unsigned char, const GemRB::Scriptable*, GemRB::STRING_FLAGS) const’ is private within this context
[build]   345 |    displaymsg->DisplayStringName( tr->textStrRef, DMC_DIALOGPARTY, speaker, STRING_FLAGS::SOUND | STRING_FLAGS::SPEECH | STRING_FLAGS::ALLOW_ZERO);
[build]       |   

[build]~/gemrb-myfork-git/gemrb/core/Scriptable/Actor.cpp:3319:65: error: ‘void GemRB::DisplayMessage::DisplayStringName(GemRB::String, unsigned char, const GemRB::Scriptable*) const’ is private within this context
[build]  3319 |    displaymsg->DisplayStringName(std::move(msg), DMC_WHITE, this);
[build]       |  
```

Reply:
Well yeah, I commented only on that one.